### ISSUE TITLE
will this work in linux

### ISSUE THREAD

**Issue by fringd** [coreyti/meltdown#2]
if so how? pow.cx only works for mac, and there's something called 'hoof' that does not work for me :/

**Comment by coreyti** [comment]
@fringd, great question!  i suspect not, at this time.  at its core, `meltdown` is backed by [rubydns](https://github.com/ioquatix/rubydns), which i believe to be cross-platform.  however, meltdown registers a "resolver" at `/etc/resolver/dev`, which may be specific to OS X.

if you try it out, please do let me know how things go.  note that the installation & execution is a bit picky at the moment.  be sure you are using "system" ruby (e.g., if using rvm first run `rvm use system`) when installing and starting/stopping `meltdown`.

@ioquatix, any thoughts?

**Comment by fringd** [comment]
okay i can't really tell if it is working, there isn't actually any usage, which makes it hard to test :)

that said, i think the thing to do in linux is edit /etc/nsswitch.conf which is what the hoof gem is trying to do...

so i'm guessing it doesn't work with linux now...

**Comment by coreyti** [comment]
@fringd right... regarding the "how":  `meltdown` for now simply resolves `*.dev` domains to localhost.  it does not provide an http server, etc., like `pow`.  so, in order to get things running under `meltdown`, you need something like apache or nginx to handling serving of your various `*.dev` domains.

for example, i use `nginx` (ssl-enabled, another reason why i had to give up on `pow`) to proxy to my `rails` applications running with `foreman` and `unicorn`.  i configure unicorn to listen on a unix socket and `nginx` to proxy to that. then i start up foreman using `foreman start -p 0`.  with the `-p 0` option, foreman picks a random port, so i can have multiple apps running at once without worrying about port collision.

i've added an issue to update the docs with usage.  will also take a look at linux usage when i can.  thanks for asking!

**Comment by ioquatix** [comment]
Most operating systems provide methods to register DNS resolvers.

On linux its /dev/resolv.conf - and you'd need super user privileges to edit it.

$ cat resolv.conf 
search sitehost.co.nz
nameserver 120.138.18.100
nameserver 120.138.22.100

You can simply add "nameserver 127.0.0.1" to that file and you will get a similar behaviour.

**Comment by coreyti** [comment]
@ioquatix great, thanks!

**Comment by fringd** [comment]
ubuntu already has dnsmasq running at 127.0.0.1, so that's problematic, there is some voodoo to do with /etc/resolver, but it seemed too dense for my simple and stupid needs...

in the end i just edited /etc/hosts with 127.0.0.1 myapp.dev myotherapp.dev etc
and run pound which is a light http proxy redirecting localhost:80 to localhost:3000, (or whatever) with this cfg:

## /etc/pound/pound.cfg

```
ListenHTTP
        Address 127.0.0.1
        Port    80

        ## allow PUT and DELETE also (by default only GET, POST and HEAD)?:
        xHTTP           0

        Service
                BackEnd
                        Address 127.0.0.1
                        Port    3000
                End
        End
End
```

if this gem ever functions as a simple http proxy and can wrangle the voodoo with /etc/resolvconf/resolv.conf.d/tail on ubuntu, it would be awesome... manually adding to /etc/hosts and pound.cfg as I add more apps is not the best
